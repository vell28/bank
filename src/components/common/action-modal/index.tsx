import React from 'react';
import { useTranslation } from 'react-i18next';

import { Portal } from '@components/portal';
import { PopupDialog } from '@components/popup/popup-overlay';

import {
  PopupTitle, PopupDescription, ActionBLock, PopupAction
} from './elements';

import { ActionBtn } from '../button/action/elements';

interface IPopupProps {
  title: string;
  description: string;
  onOk?: () => void;
  onCancel: () => void;
  onReject?: () => void;
  actionText: string;
  isShown?: boolean;
}

export const PopupBlockCard: React.FC<IPopupProps> = ({
  isShown = false,
  title,
  description,
  onOk,
  onReject,
  onCancel,
  actionText,
}) => {
  const { t } = useTranslation();
  const onActionClick = onOk || onReject || null;
  const isReject = !!onReject;
  return (
    <Portal>
      <PopupDialog isShown={isShown} onCancel={onCancel}>
        <PopupTitle>{title}</PopupTitle>
        <PopupDescription>{description}</PopupDescription>
        <ActionBLock>
          <ActionBtn onClick={onCancel}>{t('Cancel')}</ActionBtn>
          {onActionClick && (
            <PopupAction onClick={onActionClick} isReject={isReject}>
              {t(actionText)}
            </PopupAction>
          )}
        </ActionBLock>
      </PopupDialog>
    </Portal>
  );
};
