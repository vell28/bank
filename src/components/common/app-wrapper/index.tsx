import React, { PureComponent } from 'react';
import { equals } from 'ramda';

import { MainContainer } from './elements';

const HALF_OFFSET = 13;
const START_DELAY = 400;

interface IAppContainerProps {
  children: React.ReactNode;
}

export class AppWrapper extends PureComponent<IAppContainerProps> {
  public state: any = {
    percent: -1,
  };

  private container: any;

  private addTimeoutId: number | null = null;

  public componentDidMount() {
    this.setState({ percent: 0 });
  }

  public componentDidUpdate() {
    const { percent } = this.state;
    const newPercent = this.getPercent();
    if (!this.matchPercent(percent, newPercent) && this.container) {
      // eslint-disable-next-line react/no-did-update-set-state
      this.setState({ percent: newPercent });
      this.addTimeoutId = setTimeout(() => {
        this.container.classList.remove('animation');
        this.container.classList.add('animation');
      }, START_DELAY);
    }
  }

  public componentWillUnmount() {
    if (this.addTimeoutId) {
      clearTimeout(this.addTimeoutId);
    }
  }

  private getPercent() {
    if (!this.container) {
      return 0;
    }
    const activeAccount = document.querySelector('.active-account');
    const linkTop = (activeAccount && activeAccount.getBoundingClientRect().top) || 0;
    const containerTop = this.container.getBoundingClientRect().top || 0;
    return ((linkTop + HALF_OFFSET - containerTop) / this.container.offsetHeight) * 100;
  }

  private matchPercent = (prev: number, next: number): boolean => {
    return equals(Math.floor(prev), Math.floor(next));
  };

  public render() {
    const { percent } = this.state;
    const { children } = this.props;
    return (
      <MainContainer ref={(div: any) => (this.container = div)} percent={percent}>
        {children}
      </MainContainer>
    );
  }
}
