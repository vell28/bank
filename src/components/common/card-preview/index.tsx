import React from 'react';
import { equals } from 'ramda';

import { useTranslation } from 'react-i18next';
import { getMaskedCardNumber, formatCardDate } from 'utils/card';

import { CardProductType } from 'models/operations/card-settings/entities';
import { AccountStatusType, AccountStatus } from 'models/organizations/entities';
import {
  CardBox,
  AdaptiveBox,
  InnerBox,
  CardNumber,
  CardHolderName,
  CardExpiredDate,
  BlockBox,
  RoundBox,
} from './elements';

export interface ICardPreviewProps {
  productType?: CardProductType;
  cardNumber?: string;
  name: string;
  expireAt?: string;
  status: AccountStatusType;
  fontSize?: string;
}

export const CardPreview: React.FC<ICardPreviewProps> = ({
  productType = 'MAIN',
  cardNumber = '',
  name,
  expireAt = '',
  fontSize = '10',
  status,
}) => {
  const formattedDate = formatCardDate(expireAt);
  const masked = getMaskedCardNumber(cardNumber);
  const isActive = equals(status, AccountStatus.ACTIVE);
  const { t } = useTranslation();
  return (
    <CardBox fontSize={fontSize}>
      <AdaptiveBox>
        <InnerBox type={productType} isBlocked={!isActive}>
          <CardNumber key={cardNumber}>{masked}</CardNumber>
          <CardHolderName>{name}</CardHolderName>
          <CardExpiredDate>{formattedDate}</CardExpiredDate>
        </InnerBox>
      </AdaptiveBox>
      {!isActive && (
        <BlockBox>
          <RoundBox>+</RoundBox>
          <span>{t('card is blocked')}</span>
        </BlockBox>
      )}
    </CardBox>
  );
};
