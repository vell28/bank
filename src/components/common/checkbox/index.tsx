import React from 'react';

import { CheckboxBox, CheckMark, CheckContent } from './elements';

interface ICheckboxProps {
  isChecked: boolean;
  children?: React.ReactNode;
  onCheck: (isChecked: boolean) => void;
  size?: number;
}
export const CheckBox: React.FC<ICheckboxProps> = ({
  isChecked, children, size = 20, onCheck
}) => {
  const onCheckBoxClick = () => onCheck(!isChecked);
  return (
    <CheckboxBox onClick={onCheckBoxClick} isChecked={isChecked} size={size}>
      <CheckMark size={size} />
      <CheckContent>{children}</CheckContent>
    </CheckboxBox>
  );
};
