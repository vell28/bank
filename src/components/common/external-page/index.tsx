import React, { useEffect } from 'react';

import { Iframe } from './elements';

interface IExternalPageProps {
  url: string;
  onMessage: (message: string) => void;
}

export const ExternalPage: React.FC<IExternalPageProps> = ({ url, onMessage }) => {
  useEffect(() => {
    const handler = (e: MessageEvent) => {
      // TODO check e.origin
      onMessage(e.data);
    };
    window.addEventListener('message', handler);
    return () => {
      window.removeEventListener('message', handler);
    };
  }, [onMessage]);
  return (
    <>
      <Iframe src={url} />
    </>
  );
};
