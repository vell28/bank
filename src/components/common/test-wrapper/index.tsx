import React from 'react';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components';
import { mainDarkTheme } from 'modules/themes';
import { store } from 'modules/store';

interface ITestWrapper {
  children: React.ReactNode;
}

export const TestWrapper: React.FC<ITestWrapper> = ({ children }) => (
  <Provider store={store}>
    <ThemeProvider theme={mainDarkTheme}>
      <>{children}</>
    </ThemeProvider>
  </Provider>
);
