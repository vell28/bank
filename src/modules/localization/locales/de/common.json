{
  "Bank": "Bank",
  "banking": "Bankwesen",
  "bonuses": "Bonus",
  "investments": "Investition",
  "cryptotrading": "AK Crypto",
  "insurance": "Versicherung",
  "business": "Business",
  "profile": "Profil",
  "account": "Konto",
  "history": "geschichte",
  "done": "bereit",
  "new account": "Konto hinzufügen",
  "The beneficiarys bank": "Bank des Empfängers",
  "BIC": "Bank des Empfängers",
  "PHYSICAL": "personal",
  "JURIDICAL": "business",
  "sign in": "anmelden",
  "new customer": "Neukunde",
  "support": "Unterstützung",
  "Choose you language:": "Wählen Sie Sprache:"
}