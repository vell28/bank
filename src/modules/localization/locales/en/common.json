{
  "Bank": "Bank1",
  "banking": "Banking",
  "bonuses": "Bonuses",
  "investments": "investments",
  "cryptotrading": "AK Crypto",
  "insurance": "Insurance",
  "business": "Business",
  "profile": "Profile",
  "account": "account",
  "history": "history",
  "done": "done",
  "new account": "new account",
  "The beneficiarys bank": "The beneficiary's bank",
  "BIC": "BIC",
  "PHYSICAL": "personal",
  "JURIDICAL": "business",
  "Choose you language:": "Choose you language:",
  "new customer:": "new customer",
  "sign in:": "sign in:",
  "support": "support"
}
