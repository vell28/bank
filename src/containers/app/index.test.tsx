import React from 'react';
import { MemoryRouter } from 'react-router';

import { render } from 'enzyme';
import toJson from 'enzyme-to-json';

import { TestWrapper } from 'components/common/test-wrapper';
import App from '.';

describe('<App /> tests', () => {
  test('Should success render <App />', () => {
    const component: any = render(
      <TestWrapper>
        <MemoryRouter>
          <App />
        </MemoryRouter>
      </TestWrapper>,
    );
    expect(toJson(component)).toMatchSnapshot();
  });
});
