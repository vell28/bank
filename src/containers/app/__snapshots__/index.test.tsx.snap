// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> tests Should success render <App /> 1`] = `
Array [
  <header
    class="sc-pzMyG cnRxKu"
  >
    <div
      class="sc-pYA-dN ifxZWQ"
    >
      <div
        class="sc-oUcyK kUELzh"
      >
        <div
          class="sc-pbYBj blcahG"
        >
          <a
            class="sc-pzYib irdyRE"
            href="/"
          >
            <img
              alt="logo"
              src="logo.svg"
            />
          </a>
        </div>
        <nav
          class="sc-pkUbs cFdKPd"
        >
          <ul
            class="sc-psQdR bvhAnH"
          >
            <li
              class="sc-qPlga lpcyjb"
            >
              <a
                class="sc-pItiW lfpNxB"
                data-active="true"
                href="/"
              >
                Banking
              </a>
            </li>
            <li
              class="sc-qPlga lpcyjb"
            >
              <a
                class="sc-pItiW bzyDsu"
                data-active="false"
                href="/bonuses"
              >
                Bonuses
              </a>
            </li>
            <li
              class="sc-qPlga lpcyjb"
            >
              <a
                class="sc-pItiW bzyDsu"
                data-active="false"
                href="/business"
              >
                Business
              </a>
            </li>
            <li
              class="sc-qPlga lpcyjb"
            >
              <a
                class="sc-pQSRh hyqOgF"
                href="https://frontend.stage.ak-ecosystem.com/"
                target="_blank"
              >
                AK Crypto
              </a>
            </li>
            <li
              class="sc-qPlga lpcyjb"
            >
              <a
                class="sc-pItiW bzyDsu"
                data-active="false"
                href="/insurance"
              >
                Insurance
              </a>
            </li>
            <li
              class="sc-qPlga lpcyjb"
            >
              <a
                class="sc-pItiW bzyDsu"
                data-active="false"
                href="/profile"
              >
                Profile
              </a>
            </li>
          </ul>
        </nav>
        <div
          class="sc-qXhiz dlykSn"
        >
          <div
            class="sc-pZnSc fWJhPK"
          >
            <div
              class="sc-oUoif kSFiEX"
            >
              en
              <i
                class="fas fa-chevron-down"
              />
            </div>
            <div
              class="sc-pcJja ijAyKj"
            >
              <span>
                en
              </span>
              <span>
                de
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </header>,
  <div
    class="sc-qZtCU vLkXa"
  >
    <main
      class="sc-pHIBf hcryLZ"
    >
      <aside
        class="sc-ptSuy euIclc"
      >
        <div
          class="sc-paWVw kmivto"
        >
          <div
            class="sc-pjTqr jgLscH"
          >
            <button
              class="sc-oTaid fJiNEa"
              data-active="false"
              rotate="0"
            >
              accounts
            </button>
          </div>
          <div
            class="sc-qQlgh lbVPQl"
            radius="25"
          >
            <div
              class="sc-prQdK bVkINF"
              radius="25"
            >
              <div
                class="sc-qYsMX iMmCpB"
                data-overflow="true"
              >
                <div
                  class="sc-pKLCU jIUMiD"
                >
                  <div
                    class="sc-pTHXP lpvcrT"
                  >
                    <span>
                      personal
                    </span>
                  </div>
                  <div
                    class="sc-pTHXP xhpDa"
                  >
                    <span>
                      Business
                    </span>
                  </div>
                </div>
                <div
                  class="sc-pCOPB iWeFbB"
                >
                  <span>
                    done
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-qOiPt heACUx"
        >
          <div
            class="sc-qYsMX dyIjFz"
          >
            <div
              class="sc-pKLCU jIUMiD"
            />
          </div>
          <div
            class="sc-qWfCM dJAqyo"
          >
            <a
              class="sc-fznLPX eydJxS"
              href="/create-account"
            >
              new account
            </a>
          </div>
        </div>
      </aside>
      <div
        class="sc-pQEbo kEbPXp"
      >
        <section
          class="sc-qPwPv kUlfru"
        >
          <div
            class="sc-qXRQq gwPFEl"
          >
            <button
              class="sc-ptDSg eQQejW"
            >
              <svg
                height="12"
                width="12"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.263 6L12 10.737 10.737 12 6 7.263 1.263 12 0 10.737 4.737 6 0 1.263 1.263 0 6 4.737 10.737 0 12 1.263z"
                />
              </svg>
            </button>
          </div>
        </section>
        <div
          class="sc-pliRl bgDcUI"
        >
          <img
            alt="keyhole"
            src="icon-keyhole.svg"
          />
        </div>
      </div>
      <aside
        class="sc-pbKro sc-pkfsj bTKahU"
      >
        <div
          class="sc-paWVw kmivto"
        >
          <div
            class="sc-pjTqr jgLscH"
          >
            <button
              class="sc-oTaid fJiNEa"
              data-active="false"
              rotate="0"
            >
              history
            </button>
          </div>
          <div
            class="sc-qQlgh lbVPQl"
            radius="25"
          >
            <div
              class="sc-prQdK bVkINF"
              radius="25"
            >
              <div
                class="sc-qWSGh jTNOhl"
                id="scrollableDiv"
              >
                <div
                  class="sc-fzoLsD foCwer"
                >
                  <svg
                    height="26"
                    viewBox="0 0 26 26"
                    width="26"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g
                      fill="#011F3D"
                      fill-rule="nonzero"
                    >
                      <path
                        d="M12.8.25a.5.5 0 0 1 .5.5v4.82a.5.5 0 0 1-1 0V.75a.5.5 0 0 1 .5-.5z"
                      />
                      <path
                        d="M12.8 19.53a.5.5 0 0 1 .5.5v4.82a.5.5 0 0 1-1 0v-4.82a.5.5 0 0 1 .5-.5z"
                        opacity=".4"
                      />
                      <path
                        d="M9.435 6.972a.496.496 0 0 1-.683-.183l-2.41-4.175a.5.5 0 0 1 .866-.5l2.41 4.175a.5.5 0 0 1-.183.683z"
                        opacity=".9"
                      />
                      <path
                        d="M19.075 23.669a.496.496 0 0 1-.683-.183l-2.41-4.175a.5.5 0 0 1 .866-.5l2.41 4.175a.5.5 0 0 1-.183.683z"
                        opacity=".3"
                      />
                      <path
                        d="M6.972 9.435a.5.5 0 0 1-.683.183l-4.175-2.41a.5.5 0 0 1 .5-.866l4.175 2.41a.5.5 0 0 1 .183.683z"
                        opacity=".8"
                      />
                      <path
                        d="M23.669 19.075a.5.5 0 0 1-.683.183l-4.175-2.41a.5.5 0 1 1 .5-.866l4.175 2.41a.5.5 0 0 1 .183.683z"
                        opacity=".2"
                      />
                      <path
                        d="M5.57 12.3a.5.5 0 0 1 0 1H.75a.5.5 0 0 1 0-1h4.82z"
                        opacity=".7"
                      />
                      <path
                        d="M24.85 12.3a.5.5 0 0 1 0 1h-4.82a.5.5 0 0 1 0-1h4.82z"
                        opacity=".1"
                      />
                      <path
                        d="M6.789 16.848l-4.175 2.41a.496.496 0 0 1-.683-.183.5.5 0 0 1 .183-.683l4.175-2.41a.5.5 0 1 1 .5.866z"
                        opacity=".6"
                      />
                      <path
                        d="M23.486 7.208l-4.175 2.41a.497.497 0 0 1-.683-.183.5.5 0 0 1 .183-.683l4.175-2.41a.5.5 0 1 1 .5.866z"
                        opacity=".05"
                      />
                      <path
                        d="M9.618 19.311l-2.41 4.175a.5.5 0 1 1-.866-.5l2.41-4.175a.5.5 0 1 1 .866.5z"
                        opacity=".5"
                      />
                      <path
                        d="M19.258 2.614l-2.41 4.175a.5.5 0 1 1-.866-.5l2.41-4.174a.5.5 0 1 1 .866.5z"
                        opacity=".01"
                      />
                    </g>
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </div>
      </aside>
    </main>
  </div>,
]
`;
